src/main/java/com/nike/cerberus/service/AuthTokenService.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2017 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.service;
+
+import com.nike.cerberus.PrincipalType;
+import com.nike.cerberus.dao.AuthTokenDao;
+import com.nike.cerberus.domain.CerberusAuthToken;
+import com.nike.cerberus.record.AuthTokenRecord;
+import com.nike.cerberus.util.AuthTokenGenerator;
+import com.nike.cerberus.util.DateTimeSupplier;
+import com.nike.cerberus.util.RandomString;
+import com.nike.cerberus.util.TokenHasher;
+import com.nike.cerberus.util.UuidSupplier;
+import org.apache.commons.lang3.StringUtils;
+import org.mybatis.guice.datasource.dbcp.PerUserDefaultAutoCommit;
+import org.mybatis.guice.transactional.Transactional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.mybatis.guice.transactional.Isolation.READ_UNCOMMITTED;
+
+/**
+ * Service for handling authentication tokens.
+ */
+public class AuthTokenService {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final UuidSupplier uuidSupplier;
+    private final TokenHasher tokenHasher;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final AuthTokenDao authTokenDao;
+    private final DateTimeSupplier dateTimeSupplier;
+
+    @Inject
+    public AuthTokenService(UuidSupplier uuidSupplier,
+                            TokenHasher tokenHasher,
+                            AuthTokenGenerator authTokenGenerator,
+                            AuthTokenDao authTokenDao,
+                            DateTimeSupplier dateTimeSupplier) {
+
+        this.uuidSupplier = uuidSupplier;
+        this.tokenHasher = tokenHasher;
+        this.authTokenGenerator = authTokenGenerator;
+        this.authTokenDao = authTokenDao;
+        this.dateTimeSupplier = dateTimeSupplier;
+    }
+
+    @Transactional
+    public CerberusAuthToken generateToken(String principal,
```

Comment:
Do we want Hystrix metrics around these methods? It seems like before we were wrapping the VaultClient with Hystrix

Reply:
Todd has a ticket in the backlog or on his desk for Hystrix and RDS. Right now we don't have that integration